### PR TITLE
chore: adds school and trust authorization attributes

### PR DESCRIPTION
**Pull request by jrabbott**
### Context
Authorization required CFP and budget and forecasting - whereby school and trusts can only access there own data.

### Change proposed in this pull request
Adds two authorization attributes to validate user claims against route values

### Guidance to review 
Claims to be set as part of DSI integration

### Checklist
- [x] ~~Work items have been linked [(use AB#)](https://learn.microsoft.com/en-us/azure/devops/boards/github/link-to-from-github?view=azure-devops#use-ab-to-link-from-github-to-azure-boards-work-items)~~
- [x] Your code builds clean without any errors or warnings
- [x] You have run all unit/integration tests and they pass
- [x] Your branch has been rebased onto main
- [x] You have tested by running locally

### CODE REVIEW

File: web/src/Web.App/Attributes/SchoolAuthorizationAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+namespace Web.App.Attributes;
+
+[AttributeUsage(AttributeTargets.Class)]
+public class SchoolAuthorizationAttribute : AuthorizeAttribute, IAuthorizationFilter
+{
+    public void OnAuthorization(AuthorizationFilterContext context)
+    {
+        var urn = context.RouteData.Values["urn"]?.ToString();
+
+        var isValid = context.HttpContext.User.Claims.Any(c => urn != null && c.Type == "Schools" && c.Value.Contains(urn));
```
Comment (WolfyUK) [inline web/src/Web.App/Attributes/SchoolAuthorizationAttribute.cs:14]: Do we need null checks on `User.Claims` always or `Claim.Value` or is this safe enough as-is?

File: web/src/Web.App/Attributes/TrustAuthorizationAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+namespace Web.App.Attributes;
+
+[AttributeUsage(AttributeTargets.Class)]
+public class TrustAuthorizationAttribute : AuthorizeAttribute, IAuthorizationFilter
+{
+    public void OnAuthorization(AuthorizationFilterContext context)
+    {
+        var companyNumber = context.RouteData.Values["companyNumber"]?.ToString();
+
+        var isValid = context.HttpContext.User.Claims.Any(c => companyNumber != null && c.Type == "Trusts" && c.Value.Contains(companyNumber));
```
Comment (WolfyUK) [inline web/src/Web.App/Attributes/TrustAuthorizationAttribute.cs:14]: Do we need null checks on User.Claims or Claim.Value or is this safe enough as-is?

File: web/src/Web.App/Attributes/SchoolAuthorizationAttribute.cs (hunk 1 above)
Comment (jrabbott) [reply web/src/Web.App/Attributes/SchoolAuthorizationAttribute.cs:14]: Should be enough as-is. But we can ensure we have integration tests to cover this when we fully integrate DSI.

File: web/src/Web.App/Attributes/TrustAuthorizationAttribute.cs (hunk 2 above)
Comment (jrabbott) [reply web/src/Web.App/Attributes/TrustAuthorizationAttribute.cs:14]: Should be enough as-is. But we can ensure we have integration tests to cover this when we fully integrate DSI.

**Review by WolfyUK — APPROVED** [review]

**Pull request merged**